WebAssembly/wabt
# Fix spec imports test

* Properly handle registering an unnamed module (most recently read) in
  AST parser
* register command has optional name in spec JSON format
* Implement spectest_import_{table,memory,global}
* Implement on_register_command in wasm-interp (wires up module binding)
* Implement assert_unlinkable
* Set module's table_index and memory_index when importing table/memory
* Translate {get,set}_global index from module -> environment index
  space
* Don't close the environment's istream writer if the module read fails;
  this will destroy the output buffer. Just leave it as is, with extra
  unused data in the environment
* Add registered_module_bindings hash; this has the registered name of
  the module (not to be confused with the module name, which is internal
  only)
* Add {GET,SET}_GLOBAL opcodes to wasm_trace_pc and wasm_disassemble
* Add new trap when calling a function via call_indirect, when the table
  element hasn't been initialized (via an elem segment)
* Check the limits of tables + memories when importing
* Check the function signature when importing
* Use callback for printing errors from a host import; this lets us
  redirect the output the same way other errors are
* Add environment "marks". Similar to allocator marks, allows you to
  backup to a previous state when an operation fails (such as
  successfully reading a module that was supposed to fail)
